### DISCUSSION TITLE
Support annotation for Date axes in Jfree

### DISCUSSION THREAD

**Discussion by jmuthusa** [jfree/jfreechart#426] [Q&A]
Hi, 
1. All the Annotation APIs accepts input with double primitive. 
Question here is, does Jfree Annotations can take Date coordinates as it is input represented in the Dataset?	
XYSeries series1 = new XYSeries(...);     
series1.add(new Month(1, 2025), 200.0D);
XYTextAnnotation annotation = new XYTextAnnotation("A", new Month(1, 2025), 300);

2.  If not, do we have a workaround to display Texts at specified Date coordinates ? Have came across samples that converts the date value to double on Dataset and on the Annotations objects. The problem is the coordinates are somehow going out of range and texts are not displayed.

Thanks in Advance.

**Comment by trashgod** [comment]
Yes, annotations work with dates; some examples are seen [here](https://stackoverflow.com/search?q=user%3A230513+%5Bjfreechart%5D+annotation+date).

**Reply by jmuthusa** [reply]
Yes, that worked. Great! Thank You!
Got the index of the Date component as in the sample and set as X coordinate.
 double t0 = timeSeries.getTimePeriod(0).getFirstMillisecond() 

For a Multi-Series dataset, how do we position the Annotation on specific bar.
![image](https://github.com/user-attachments/assets/426a5474-57d4-47b5-9bbf-ad0ad45afe54)

As in the image , would expect "A" to be on red bar and "B" on blue bar.

**Comment by trashgod** [comment]
I'm not sure; you might experiment with the `RegularTimePeriod` methods involving _first_, _middle_ and _last_ millisecond, as well as `TimePeriodAnchor`.

In addition, this related [example](https://gist.github.com/trashgod/a3f7581417db19bf5b6dc52d3647a2f9) illustrates another way to interpolate values on the time axis.
